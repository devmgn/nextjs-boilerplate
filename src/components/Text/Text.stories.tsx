import { faker } from '@faker-js/faker/locale/ja';
import { Text } from '.';
import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Text> = {
  title: 'components/Text',
  component: Text,
  args: {
    children: faker.lorem.paragraph(),
  },
};

export default meta;
type Story = StoryObj<typeof Text>;

export const Default: Story = {};
