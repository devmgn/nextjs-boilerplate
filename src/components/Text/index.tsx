import { Text as RuiText } from '@radix-ui/themes';
import clsx from 'clsx';
import { text } from './style.css';
import type { TextProps } from '@radix-ui/themes/dist/cjs/components/text';

export const Text: React.FC<TextProps> = ({
  className,
  weight = 'regular',
  ...props
}) => {
  return (
    <RuiText {...props} weight={weight} className={clsx(text, className)} />
  );
};
