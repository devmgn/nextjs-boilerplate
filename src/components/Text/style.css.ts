import { style } from '@vanilla-extract/css';

export const text = style({
  fontSize: 14,
  lineHeight: 1.5,
});
