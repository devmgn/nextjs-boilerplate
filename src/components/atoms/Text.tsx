import styled from 'styled-components';

const Text = styled.p`
  font-size: 16px;
  line-height: 24px;
  color: #4c4d4d;
`;

export default Text;
