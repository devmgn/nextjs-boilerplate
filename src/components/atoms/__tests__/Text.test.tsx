import { render } from '@testing-library/react';
import Text from '../Text';

describe('Text', () => {
  test('スナップショットテスト', () => {
    const { asFragment } = render(<Text>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});
