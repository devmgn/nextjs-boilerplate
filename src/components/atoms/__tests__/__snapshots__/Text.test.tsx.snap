// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text スナップショットテスト 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 16px;
  line-height: 24px;
}

<p
    class="c0"
  >
    children
  </p>
</DocumentFragment>
`;
